hal/src/rtl872x/exflash_hal.cpp

```diff
@@ -554,6 +558,10 @@ static int perform_write(uintptr_t addr, const uint8_t* data, size_t size) {
 int hal_exflash_init(void) {
     // ExFlashLock lk;
     // FLASH_ClockSwitch(BIT_SHIFT_FLASH_CLK_XTAL, _FALSE);
+    uint32_t CPUID = (SCB->CPUID & SCB_CPUID_PARTNO_Msk) >> SCB_CPUID_PARTNO_Pos;
+    if (CPUID == 0xD20 ) {
+        isKM0 = true;
+    }
```

Comment:
I tried reading `SCB->CPUID` each time we considered shifting the Status register value, but things slowed down dramatically. Detecting the KM0 core or not should only need to be done once on boot